File: 5a/Client/src/HelloWorldClientHostedService.fs
module Client.HelloWorldClientHostedService

open System.Threading
open System.Threading.Tasks
open Microsoft.Extensions.Hosting

type Service() =
    interface IHostedService with
        member _.StartAsync(cancellationToken: CancellationToken) : Task = Task.CompletedTask
        member _.StopAsync(cancellationToken: CancellationToken) : Task = Task.CompletedTask

File: 3c/Client/src/HelloWorldClientHostedService.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Interfaces;
using Microsoft.Extensions.Hosting;
using Orleans;

namespace Client
{
    public class HelloWorldClientHostedService : IHostedService
    {
        private readonly IClusterClient _clusterClient;

        public HelloWorldClientHostedService(IClusterClient clusterClient)
        {
            _clusterClient = clusterClient;
        }

        public async Task StartAsync(CancellationToken cancellationToken)
        {
            var helloWorldGrain = _clusterClient.GetGrain<IHelloWorld>(0);
            Console.WriteLine($"{await helloWorldGrain.Factorial(3)}");
        }

        public Task StopAsync(CancellationToken cancellationToken) => Task.CompletedTask;
    }
}